src/repr/src/row.rs

```diff
@@ -832,6 +860,72 @@ impl RowPacker {
         res
     }
 
+    /// Convenience function to construct an array from an iter of `Datum`s.
+    ///
+    /// Returns an error if the number of elements in `iter` does not match
+    /// the cardinality of the array as described by `dims`, or if the
+    /// number of dimensions exceeds [`MAX_ARRAY_DIMENSIONS`]. If an error
+    /// occurs, the packer's state will be unchanged.
+    pub fn push_array<'a, I, D>(
+        &mut self,
+        dims: &[ArrayDimension],
+        iter: I,
+    ) -> Result<(), InvalidArrayError>
+    where
+        I: IntoIterator<Item = D>,
+        D: Borrow<Datum<'a>>,
+    {
+        // Arrays are encoded as follows.
+        //
+        // u8      ndims
+        // usize   dim_0 lower bound
+        // usize   dim_0 length
+        // ...
+        // usize   dim_n lower bound
+        // usize   dim_n length
+        // usize   element data size in bytes
+        // u8      element data, where elements are encoded in row-major order
+
+        if dims.len() >= usize::from(MAX_ARRAY_DIMENSIONS) {
+            return Err(InvalidArrayError::TooManyDimensions(dims.len()));
+        }
+
+        let start = self.data.len();
+        self.data.push(Tag::Array as u8);
+
+        // Write dimension information.
+        self.data
+            .push(dims.len().try_into().expect("ndims verified to fit in u8"));
```

Comment:
Since this is a public method, someone might someday erroneously use it from a different context, so I'd slightly prefer a more generic panic message, like "ndims must fit in u8".

Reply:
It's verified on L888 to fit in a u8 within this method, which is why the `expect` is phrased this way. Does that change your opinion? Still happy to try to make that more clear.

Reply:
oh ok, it's fine then. I missed that.